app/src/main/java/com/example/ithaca_transit_android_v2/views/RvAdapter.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.ithaca_transit_android_v2.views
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.example.ithaca_transit_android_v2.R
+import com.example.ithaca_transit_android_v2.models.Route
+import com.example.ithaca_transit_android_v2.states.RouteCardState
+import com.example.ithaca_transit_android_v2.states.RouteDetailViewState
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+
+class RvAdapter(val userList: ArrayList<Route>) : RecyclerView.Adapter<RvAdapter.ViewHolder>() {
+
+    //Creation of the observable object, defining that it will hold a RouteCardState
+    private val clickSubject = PublishSubject.create<RouteCardState>()
+    val clickEvent: Observable<RouteCardState> = clickSubject
+
+
+    override fun onCreateViewHolder(p0: ViewGroup, p1: Int): ViewHolder {
+        val v = LayoutInflater.from(p0?.context).inflate(R.layout.routecards, p0, false)
+        return ViewHolder(v);
+    }
+
+    override fun getItemCount(): Int {
+        return userList.size
+    }
+
+    override fun onBindViewHolder(p0: ViewHolder, p1: Int) {
+
+        p0.description?.text = userList[p1].arrival.toString()
+        p0.delay?.text = userList[p1].boardInMin.toString()
```

Comment:
check to make sure that `description` and `delay` are nullable. if not, remove the `?`

Reply:
These are just place holders so that I can see the difference between each route card.